gpMgmt/bin/go-tools/agent/server_test.go

```diff
@@ -0,0 +1,180 @@
+package agent_test
+
+import (
+	"errors"
+	"net"
+	"reflect"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/greenplum-db/gp-common-go-libs/testhelper"
+	agent "github.com/greenplum-db/gpdb/gp/agent"
+	"github.com/greenplum-db/gpdb/gp/idl"
+	"github.com/greenplum-db/gpdb/gp/testutils"
+)
+
+func TestStartServer(t *testing.T) {
+	testhelper.SetupTestLogger()
+
+	t.Run("successfully starts the server", func(t *testing.T) {
+
+		credCmd := &testutils.MockCredentials{}
+
+		agentServer := agent.New(agent.Config{
+			Port:        8000,
+			ServiceName: "gp",
+			Credentials: credCmd,
+		})
+
+		errChan := make(chan error, 1)
+		go func() {
+			errChan <- agentServer.Start()
+		}()
+
+		defer agentServer.Shutdown()
+
+		select {
+		case err := <-errChan:
+			if err != nil {
+				t.Fatalf("unexpected error: %#v", err)
+			}
+		case <-time.After(1 * time.Second):
+			t.Log("server started listening")
+		}
+
+	})
+
+	t.Run("failed to start if the load credential fail", func(t *testing.T) {
+
+		credCmd := &testutils.MockCredentials{}
+		credCmd.SetCredsError("Test credential error")
+
+		agentServer := agent.New(agent.Config{
+			Port:        8001,
+			ServiceName: "gp",
+			Credentials: credCmd,
+		})
+
+		errChan := make(chan error, 1)
+		go func() {
+			errChan <- agentServer.Start()
+		}()
+		defer agentServer.Shutdown()
+
+		select {
+		case err := <-errChan:
+			if !strings.Contains(err.Error(), "Could not load credentials") {
+				t.Fatalf("want \"Could not load credentials\" but get: %q", err.Error())
```

Comment:
Alternative if you don't want to escape quotation.
```suggestion
				t.Fatalf(`Expected "Could not load credentials" but got: %q`, err.Error())
```

Reply:
In fact its common practice and easier to follow by extracting to an expected variable. Please make this change throughout the entire code base.

```go
expected := "Could not load credentials"
if !strings.Contains(err.Error(), expected) {
				t.Fatalf("expected %q to contain %q, err.Error(), expected)
}
```

Reply:
+1 ^